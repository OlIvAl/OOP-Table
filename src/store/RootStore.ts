import { FinanceActionsStore } from './Actions/FinanceActionsStore';
import { IFinanceActionModel } from './Actions/interfaces';
import { IStoreWithCurrentModel } from './Abstract/interfaces';

export class RootStore {
  financeActionsStore: IStoreWithCurrentModel<IFinanceActionModel>;

  constructor() {
    this.financeActionsStore = new FinanceActionsStore(this);
  }
}