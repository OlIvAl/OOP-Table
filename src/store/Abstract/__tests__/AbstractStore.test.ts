import { IStore } from '../interfaces';
import { ITestModel } from '../Test/interfaces';
import TestStore from '../Test/TestStore';
import TestModel from '../Test/TestModel';
import { RootStore } from '../../RootStore';

let store: IStore<ITestModel>;
let models: ITestModel[];

const rootStore = new RootStore();

beforeEach(() => {
  store = new TestStore(rootStore);
  models = [new TestModel(store), new TestModel(store), new TestModel(store)];
  store.models = models;
});

describe('AbstractStore', () => {
  describe('constructor', () => {
    it('init valid store', () => {
      expect(store).toMatchSnapshot();
    });
  });
  describe('resetStore', () => {
    it('clean models', () => {
      expect(store.models.length).toEqual(3);
      store.resetStore();
      expect(store.models.length).toEqual(0);
    });
  });
});
