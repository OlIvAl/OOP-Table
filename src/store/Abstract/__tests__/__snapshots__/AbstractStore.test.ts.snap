// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractStore constructor init valid store 1`] = `
TestStore {
  "models": Array [
    TestModel {
      "store": [Circular],
    },
    TestModel {
      "store": [Circular],
    },
    TestModel {
      "store": [Circular],
    },
  ],
  "rootStore": RootStore {
    "financeActionsStore": FinanceActionsStore {
      "rootStore": [Circular],
    },
  },
}
`;
